index.js

```diff
@@ -216,14 +218,25 @@ function request(method, url, options, callback) {
 
   var responded = false;
 
-  var req = protocols[url.protocol.replace(/\:$/, '')].request({
+  var params = {
     host: url.hostname,
     port: url.port,
     path: url.path,
     method: method,
     headers: rawHeaders,
     agent: agent
-  }, function (res) {
+  }
+
+  if (options.proxy === null || options.proxy === undefined || options.proxy === '') {
+    options.proxy = 'http_proxy' in env ? env.http_proxy : false;
+  }
+
+  if (options.proxy) {
+    var proxyAgent = new proxyAgents[url.protocol.replace(/\:$/, '')](options.proxy);
+    params.agent = proxyAgent;
```

Comment:
Given that all you need to do is override the agent, why don't we just accept an `agent` option?  That way we can avoid depending on the extra `*-proxy-agent` modules ourselves.